game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java

```diff
@@ -247,9 +246,7 @@ private void centerOnMovableUnit(final boolean selectNext) {
       if (!matchedUnits.isEmpty()) {
         drawUnitAvatarPane(t);
         newFocusedTerritory = t;
-        final Collection<List<Unit>> highlight = new ArrayList<>();
-        highlight.add(matchedUnits);
-        mapPanel.setUnitHighlight(highlight);
+        mapPanel.setUnitHighlight(Collections.singleton(matchedUnits));
```

Comment:
Is matched units guaranteed to be immutable?

Reply:
Yes, `matched` is local to the for loop, there is a `break` that comes next exiting the for loop.